packages/sanity/src/studio/components/navbar/schemaWarnings/SchemaWarningsButton.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useCallback, useMemo, useState} from 'react'
+import {
+  Button,
+  Card,
+  Heading,
+  Text,
+  Menu,
+  MenuButton,
+  Stack,
+  Dialog,
+  DialogProps,
+  Inline,
+} from '@sanity/ui'
+
+import {InfoOutlineIcon, PackageIcon, WarningOutlineIcon} from '@sanity/icons'
+import {useId} from '@reach/auto-id'
+
+import {ChangelogAccordion} from '../changelog/ChangelogAccordion'
+import {isDev} from '../../../../environment'
+import {useSchema} from '../../../../hooks'
+import {useColorScheme} from '../../../colorScheme'
+import {SchemaProblemGroups} from '../../../screens/schemaErrors/SchemaProblemGroups'
+
+export function SchemaWarningsButton() {
+  const schema = useSchema()
+  const {scheme} = useColorScheme()
+  const groupsWithWarnings =
+    schema._validation?.filter((group) =>
+      group.problems.some((problem) => problem.severity === 'warning')
+    ) || []
+
+  const popoverProps = useMemo(
+    () => ({
+      constrainSize: true,
+      portal: true,
+      scheme: scheme,
+    }),
+    [scheme]
+  )
+
+  const dialogId = useId() || 'schema-warnings-menu'
+
+  const [isDialogOpen, setDialogOpen] = useState(false)
+  const [buttonElement, setButtonElement] = useState<HTMLButtonElement | null>(null)
+  const handleOpen = useCallback(() => setDialogOpen(true), [])
+
+  const handleClose = useCallback(() => {
+    setDialogOpen(false)
+
+    if (buttonElement) {
+      buttonElement.focus()
+    }
+  }, [buttonElement])
+
+  const dialogProps: Omit<DialogProps, 'id'> = useMemo(
+    () => ({
+      footer: (
+        <Inline padding={2} space={2}>
+          <InfoOutlineIcon />
+          <Text muted size={1}>
+            Note: this message is only displayed during development
+          </Text>
+        </Inline>
+      ),
+      onClickOutside: handleClose,
+      onClose: handleClose,
+      scheme: 'light',
+    }),
+    [handleClose]
+  )
+
+  if (groupsWithWarnings.length === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <Button
+        icon={WarningOutlineIcon}
+        mode="bleed"
+        tone="caution"
+        title={`The current schema has ${groupsWithWarnings.length} warnings`}
+        onClick={handleOpen}
+        ref={setButtonElement}
+        selected={isDialogOpen}
+      />
+      {isDialogOpen && (
+        <Dialog
+          header={`Found ${groupsWithWarnings.length} warnings with the current schema`}
+          width={2}
+          {...dialogProps}
+          id={dialogId}
+        >
+          <Stack space={4} padding={3}>
```

Comment:
This `Stack` is unnecessary since the `SchemaProblemsGroup` has its once `Stack`. 

I suggest changing it to a `Box` and bump to `padding` to `4`!

```js
<Box padding={4}>
  <SchemaProblemGroups problemGroups={groupsWithWarnings} />
</Box>
```